packages/components/tree-v2/src/tree-node.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div
+    ref="node$"
+    class="el-tree-node"
+    :class="{
+      'is-expanded': expanded,
+      'is-current': current,
+      'is-focusable': !disabled,
+      'is-checked': !disabled && checked,
+    }"
+    role="treeitem"
+    tabindex="-1"
+    :aria-expanded="expanded"
+    :aria-disabled="disabled"
+    :aria-checked="checked"
+    :data-key="node?.key"
+    @click.stop="handleClick"
+    @contextmenu="handleContextMenu"
+  >
+    <div
+      class="el-tree-node__content"
+      :style="{ paddingLeft: `${(node.level - 1) * indent}px` }"
+    >
+      <el-icon
+        :class="[
+          {
+            'is-leaf': node?.isLeaf,
+            'is-hidden': hiddenExpandIcon,
+            expanded: !node?.isLeaf && expanded,
+          },
+          'el-tree-node__expand-icon',
+        ]"
+        @click.stop="handleExpandIconClick"
+      >
+        <component :is="icon" />
+      </el-icon>
+      <el-checkbox
+        v-if="showCheckbox"
+        :model-value="checked"
+        :indeterminate="indeterminate"
+        :disabled="disabled"
+        @change="handleCheckChange"
+        @click.stop
+      />
+      <el-node-content :node="node" />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, inject } from 'vue'
+import { CaretRight } from '@element-plus/icons'
+import ElIcon from '@element-plus/components/icon'
+import ElCheckbox from '@element-plus/components/checkbox'
+import ElNodeContent from './tree-node-content.vue'
+import { ROOT_TREE_INJECTION_KEY, treeNodeProps } from './virtual-tree'
+import type { RootTreeType } from './tree.type'
+
+const DEFAULT_ICON = 'caret-right'
+
+export default defineComponent({
+  name: 'ElTreeNode',
+  components: {
+    ElIcon,
+    CaretRight,
+    ElCheckbox,
+    ElNodeContent,
+  },
+  props: treeNodeProps,
+  emits: ['click', 'toggle', 'check'],
```

Comment:
We should use emits with type.
Example: https://github.com/element-plus/element-plus/blob/51cc9565898e48b77846969c09e02807cbe2b699/packages/components/button/src/button.ts#L41